tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_testcase.py

```diff
@@ -465,16 +467,28 @@ def _fetch_and_check_xds_config(self):
         # Execute the child class provided validation logic
         self.xds_config_validate(DumpedXdsConfig(self._xds_json_config))
 
+    def _print_error_list(self, flavour, errors):
+        for _, err in errors:
+            logging.error("%s: %s", flavour, self.__class__.__name__)
+            logging.error("%s", err)
+
     def run(self, result: unittest.TestResult = None) -> None:
         """Abort this test case if CSDS check is failed.
 
         This prevents the test runner to waste time on RPC distribution test,
         and yields clearer signal.
         """
+        global _first_error_printed  # pylint: disable=global-statement
+
         if result.failures or result.errors:
+            if not _first_error_printed:
+                self._print_error_list("ERROR", result.errors)
+                self._print_error_list("FAIL", result.failures)
+                _first_error_printed = True
             logging.info("Aborting %s", self.__class__.__name__)
         else:
             super().run(result)
+            logging.info("Passed test: %s", self.__class__.__name__)
```

Comment:
I think if a class has multiple tests, this will just print test class name for all of them. IRRC you should be able to use `self.id()`